pkg/virtualKubelet/forge/pods.go

```diff
@@ -15,145 +15,92 @@
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/klog/v2"
+	metricsv1beta1 "k8s.io/metrics/pkg/apis/metrics/v1beta1"
 
+	vkv1alpha1 "github.com/liqotech/liqo/apis/virtualkubelet/v1alpha1"
 	liqoconst "github.com/liqotech/liqo/pkg/consts"
-	liqonetIpam "github.com/liqotech/liqo/pkg/liqonet/ipam"
-	"github.com/liqotech/liqo/pkg/virtualKubelet"
-	apimgmt "github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection"
-	"github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection/reflectors"
 )
 
-const affinitySelector = liqoconst.TypeNode
-
-func (f *apiForger) podForeignToHome(foreignObj, homeObj runtime.Object, reflectionType string) (*corev1.Pod, error) {
-	var isNewObject bool
-
-	if homeObj == nil {
-		isNewObject = true
-
-		homeObj = &corev1.Pod{
-			TypeMeta:   metav1.TypeMeta{},
-			ObjectMeta: metav1.ObjectMeta{},
-			Spec:       corev1.PodSpec{},
-		}
-	}
-
-	foreignPod := foreignObj.(*corev1.Pod)
-	homePod := homeObj.(*corev1.Pod)
-
-	foreignNamespace, err := f.nattingTable.DeNatNamespace(foreignPod.Namespace)
-	if err != nil {
-		return nil, err
-	}
+const (
+	// PodRejectedReason -> the reason assigned to pods rejected by the virtual kubelet.
+	PodRejectedReason = "OffloadingRejected"
+)
 
-	f.forgeHomeMeta(&foreignPod.ObjectMeta, &homePod.ObjectMeta, foreignNamespace, reflectionType)
-	delete(homePod.Labels, virtualKubelet.ReflectedpodKey)
+// PodIPTranslator defines the function to translate between remote and local IP addresses.
+type PodIPTranslator func(string) string
 
-	if isNewObject {
-		homePod.Spec = f.forgePodSpec(foreignPod.Spec)
+// LocalPod forges the status of the local pod, given the remote one.
+func LocalPod(local, remote *corev1.Pod, translator PodIPTranslator, restarts int32) *corev1.Pod {
+	return &corev1.Pod{
+		ObjectMeta: *local.ObjectMeta.DeepCopy(),
+		Status:     LocalPodStatus(remote.Status.DeepCopy(), translator, restarts),
 	}
-
-	return homePod, nil
 }
 
-func (f *apiForger) podStatusForeignToHome(foreignObj, homeObj runtime.Object) *corev1.Pod {
-	homePod := homeObj.(*corev1.Pod)
-	foreignPod := foreignObj.(*corev1.Pod)
-
-	homePod.Status = foreignPod.Status
-	if homePod.Status.PodIP != "" {
-		response, err := f.ipamClient.GetHomePodIP(context.Background(),
-			&liqonetIpam.GetHomePodIPRequest{
-				ClusterID: RemoteClusterID,
-				Ip:        foreignPod.Status.PodIP,
-			})
-		if err != nil {
-			klog.Error(err)
-		}
-		homePod.Status.PodIP = response.GetHomeIP()
-		homePod.Status.PodIPs[0].IP = response.GetHomeIP()
+// LocalPodStatus forges the status of the local pod, given the remote one.
+func LocalPodStatus(remote *corev1.PodStatus, translator PodIPTranslator, restarts int32) corev1.PodStatus {
+	// Translate the relevant IPs
+	if remote.PodIP != "" {
+		remote.PodIP = translator(remote.PodIP)
+		remote.PodIPs = []corev1.PodIP{{IP: remote.PodIP}}
```

Comment:
Should this IP be translated too?

Reply:
It is, since it is assigned the one already translated the line above.